avocado/plugins/config.py

```diff
@@ -39,7 +42,32 @@ def configure(self, parser):
                                         parser=parser,
                                         long_arg='--datadir')
 
+        subcommands = parser.add_subparsers(dest='config_subcommand',
+                                            metavar='sub-command')
+
+        help_msg = 'Show a configuration reference with all registered options'
+        subcommands.add_parser('reference', help=help_msg)
```

Comment:
As the code is displaying the configs on the terminal, I would go with a more intuitive command name, like `show`, `avocado config show` feel more natural to me.

Anyway, this is not a big deal.

Reply:
@willianrampazzo IMO, `avocado config show` gives the impression that will show the current config. And it is not the case. Maybe `show-reference` but I'm trying to keep commands short.

Reply:
Reading the PR description I thought the command was supposed to show the current config options, independent from where they come. 

What happens if I execute `avocado --config /tmp/foo.conf config reference`? I imagine it will use the internal default options, plus the updated options from `/tmp/foo.conf`. This is the reason I thought `show` would make more sense initially.

Reply:
**current configuration reference guide** maybe my bad English gave that impression. Sorry. Is the "current reference guide"